peps/pep-0748.rst

```diff
@@ -0,0 +1,1574 @@
+  OpenSSL. This puts the CPython development team in the position of being
+  OpenSSL redistributors, potentially needing to ship security updates to the
+  Windows Python distributions when OpenSSL vulnerabilities are released.
+
+* macOS distributions of Python need either to be shipped with a copy
+  of OpenSSL or linked against the system OpenSSL library. Apple has formally
+  deprecated linking against the system OpenSSL library, and even if they had
+  not, that library version has been unsupported by upstream for nearly one year
+  as of the time of writing. The CPython development team has started shipping
+  newer OpenSSLs with the Python available from python.org, but this has the
+  same problem as with Windows.
+
+* Users may wish to integrate with TLS libraries other than OpenSSL for other
+  reasons, such as maintenance burden versus a system-provided implementation,
+  or because OpenSSL is simply too large and unwieldy for their platform (e.g.
+  for embedded Python). Those users are left with the requirement to use
+  third-party networking libraries that can interact with their preferred TLS
+  library or to shim their preferred library into the OpenSSL-specific
+  :mod:`ssl` module API.
+
+Additionally, the :mod:`ssl` module as implemented today limits the ability of
+CPython itself to add support for alternative TLS implementations, or remove
+OpenSSL support entirely, should either of these become necessary or useful. The
+:mod:`ssl` module exposes too many OpenSSL-specific function calls and features
+to easily map to an alternative TLS implementation.
+
+Proposal
+========
+
+This PEP proposes to introduce a few new Protocol Classes in Python 3.14 to
+provide TLS functionality that is not so strongly tied to OpenSSL. It also
+proposes to update standard library modules to use only the interface exposed by
+these protocol classes wherever possible. There are three goals here:
+
+1. To provide a common API surface for both core and third-party developers to
+   target their TLS implementations to. This allows TLS developers to provide
+   interfaces that can be used by most Python code, and allows network
+   developers to have an interface that they can target that will work with a
+   wide range of TLS implementations.
+
+1. To provide an API that has few or no OpenSSL-specific concepts leak through.
+   The :mod:`ssl` module today has a number of warts caused by leaking OpenSSL
+   concepts through to the API: the new protocol classes would remove those
+   specific concepts.
+
+1. To provide a path for the core development team to make OpenSSL one of many
+   possible TLS implementations, rather than requiring that it be present on a
+   system in order for Python to have TLS support.
+
+The proposed interface is laid out below.
+
+Interfaces
+----------
+
+There are several interfaces that require standardization. Those interfaces are:
+
+1. Configuring TLS, currently implemented by the :class:`~ssl.SSLContext` class
```

Comment:
Given the back references from the text below, it's likely clearer to use explicit bullet point numbering on this list, rather than relying on the ReST auto-numbering.